pkg/apis/v1beta1/ec2nodeclass.go

```diff
@@ -41,6 +41,9 @@ type EC2NodeClassSpec struct {
 	// +kubebuilder:validation:MaxItems:=30
 	// +required
 	SecurityGroupSelectorTerms []SecurityGroupSelectorTerm `json:"securityGroupSelectorTerms" hash:"ignore"`
+	// AssociatePublicIPAddress controls if public IP addresses are assigned to instances that are launched by the launch template.
```

Comment:
Does this make sense? the fact that we generate a launch template and is passed through aws is more of an implementation detail.
```suggestion
	// AssociatePublicIPAddress controls if public IP addresses are assigned to instances that are launched with the nodeclass.
```